source/src/AnalysisHelper.cc

```diff
@@ -0,0 +1,179 @@
+/// \file AnalysisHelper.cc
+/*
+ *
+ * AnalysisHelper.cc source template automatically generated by a class generator
+ * Creation date : mar. juil. 1 2014
+ *
+ * This file is part of DQM4HEP libraries.
+ *
+ * DQM4HEP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * based upon these libraries are permitted. Any copy of these libraries
+ * must include this copyright notice.
+ *
+ * DQM4HEP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with DQM4HEP.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Tom Coates
+ * @copyright CNRS , IPNL
+ */
+
+// -- dqm4hep headers
+#include <dqm4hep/AnalysisHelper.h>
+#include <typeinfo>
+
+namespace dqm4hep {
+
+  namespace core {
+
+    float AnalysisHelper::mean(MonitorElementPtr pMonitorElement, float percentage = 1.0)
+    {
+
+      std::string ObjectType;
+
+      float result = 0.0;
+
+      TObject *ThisObject = pMonitorElement->objectTo<TObject>();
+
+      if (pMonitorElement->type() == "TH1F" || pMonitorElement->type() == "TH1I") {  // How many types of TH1 histogram do we have to recognise? TH1F, TH1I, ...?
+	ObjectType = "TH1";
+      }
+      else if (pMonitorElement->type() == "TGraph") {
+	ObjectType = "TGraph";
+      }
+      else {
+	// Really this should be replaced with something that goes in the log
+	std::cout << "The monitor element " << pMonitorElement->name() << " of type " << pMonitorElement->type() << " could not be classified." << std::endl;
```

Comment:
Here use our logging interface : 
```cpp
dqm_error( "Message" );
```

Reply:
I'm trying to have the error message give both the incorrect argument *and* the name of the qtest where it occurred, doing it like this:
    dqm_error("{0} : Unrecognised test type: {1}", name, m_property);
and it gives me `error: invalid use of non-static member function`, so it seems like I'm using it incorrectly. Am I using the function wrong, or is it an issue with passing the `name` variable to the dqm_error macro?

Reply:
Yes, if you want to access the QTest name, you should use `name()` not `name`, which is, in the latter case, considered as a static member function (which is not, as the compiler says).